src/cargo/ops/cargo_rustc/context.rs

```diff
@@ -365,6 +369,20 @@ impl<'a, 'cfg> Context<'a, 'cfg> {
                                     target.is_example() ||
                                     profile.test;
 
+                // If this dependency is only available for certain platforms,
+                // make sure we're only enabling it for that platform.
+                let is_platform_same = conds.iter().any(|cond| {
+                    match (kind, cond.platform()) {
+                        (Kind::Host, Some(platform)) => {
+                            platform == self.config.rustc_info().host
+                        },
+                        (Kind::Target, Some(platform)) => {
+                            platform == self.target_triple
+                        },
+                        _ => true
```

Comment:
Could you change this case to `(_, None)` to ensure we get the help of the exhaustiveness checker? e.g. it's clear that no requirement means it should always be linked.